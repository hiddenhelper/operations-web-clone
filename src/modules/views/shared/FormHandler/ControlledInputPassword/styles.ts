import { createStyles, makeStyles } from '@material-ui/core/styles';
import { STYLE } from '../../../../../constants';

export const useStyles = makeStyles(theme =>
  createStyles({
    inputPasswordWrapper: {
      '& .MuiInputBase-input': {
        paddingLeft: 0,
        position: 'relative',
        top: 1,
      },
    },
    inputPasswordBackground: {
      '& .MuiOutlinedInput-root': {
        backgroundColor: STYLE.COLOR.INPUT_BACKGROUND,
      },
      '& .MuiOutlinedInput-input': {
        backgroundColor: 'transparent',
      },
    },
    inactiveIcon: {
      '& path': {
        fill: STYLE.COLOR.INACTIVE,
      },
    },
    errorIcon: {
      '& path': {
        fill: STYLE.COLOR.ERROR_ICON,
      },
    },
    filledInput: {
      '& path': {
        fill: STYLE.COLOR.SECONDARY_DARKER,
      },
    },
  })
);
