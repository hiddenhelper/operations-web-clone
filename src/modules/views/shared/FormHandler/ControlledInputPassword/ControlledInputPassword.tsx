import React, { memo, useState, useCallback } from 'react';

import InputAdornment from '@material-ui/core/InputAdornment';
import LockIcon from '@material-ui/icons/Lock';
import IconButton from '@material-ui/core/IconButton';
import TextField from '@material-ui/core/TextField';

import ControlledInput from '../ControlledInput';

import { VisibilityOn, VisibilityOff } from '../../../../../constants';

import { useStyles as inputStyles } from '../ControlledInput/styles';
import { useStyles } from './styles';
import { getConditionalDefaultValue } from '../../../../../utils/generalUtils';

export interface IControlledInputPasswordProps {
  fieldName: string;
  value: string;
  label: string;
  error: string;
  rules: any;
  placeholder?: string;
  dataTestId?: string;
  styleClass?: any;
  onChangeHandler: (value) => void;
  onBlurValidate: (value) => void;
}

const ControlledInputPassword = ({
  fieldName,
  value,
  label,
  error,
  rules,
  placeholder,
  dataTestId,
  styleClass = '',
  onChangeHandler,
  onBlurValidate,
}: IControlledInputPasswordProps) => {
  const classes = useStyles();
  const inputClasses = inputStyles();

  const [showPassword, setShowPassword] = useState<boolean>(false);

  const handleClickShowPassword = useCallback(() => {
    setShowPassword(prev => !prev);
  }, []);

  const handleMouseDownPassword = useCallback((event: React.MouseEvent<HTMLButtonElement>) => {
    event.preventDefault();
  }, []);

  return (
    <ControlledInput styleClass={`${inputClasses.loginLabels} ${styleClass}`} label={label} required={true} dataTestId="confirm-password-wrapper">
      <TextField
        name={fieldName}
        autoComplete="off"
        variant="outlined"
        fullWidth={true}
        placeholder={placeholder}
        type={showPassword ? 'text' : 'password'}
        className={inputClasses.textField}
        aria-describedby="confirm-password-password"
        required={rules.required}
        value={value}
        onChange={onChangeHandler}
        onBlur={onBlurValidate}
        error={!!error}
        inputProps={{
          'data-testid': dataTestId,
        }}
        InputProps={{
          startAdornment: (
            <InputAdornment
              className={`${value.length > 0 && !error ? classes.filledInput : ''}  ${!!error ? classes.errorIcon : ''}  ${inputClasses.loginIcons}`}
              position="start"
            >
              <LockIcon />
            </InputAdornment>
          ),
          endAdornment: (
            <InputAdornment position="end">
              <IconButton
                data-testid="toggle-visibility-button"
                aria-label="toggle-visibility"
                onClick={handleClickShowPassword}
                onMouseDown={handleMouseDownPassword}
                edge="end"
              >
                {showPassword ? (
                  <VisibilityOn
                    className={` ${classes.inactiveIcon} ${getConditionalDefaultValue(value.length > 0, classes.filledInput, classes.inactiveIcon)}`}
                  />
                ) : (
                  <VisibilityOff
                    className={` ${classes.inactiveIcon} ${getConditionalDefaultValue(value.length > 0, classes.filledInput, classes.inactiveIcon)}`}
                  />
                )}
              </IconButton>
            </InputAdornment>
          ),
        }}
      />
    </ControlledInput>
  );
};

export default memo(ControlledInputPassword);
