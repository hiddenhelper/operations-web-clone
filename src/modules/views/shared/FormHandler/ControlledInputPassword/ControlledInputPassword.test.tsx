import React from 'react';
import { render, act, fireEvent } from '@testing-library/react';
import ControlledInputPassword, { IControlledInputPasswordProps } from './ControlledInputPassword';

describe('Controlled Input Password', () => {
  let props: IControlledInputPasswordProps = {
    fieldName: 'password',
    label: 'Current Password',
    value: '',
    error: null,
    rules: {},
    placeholder: 'Current Password',
    dataTestId: 'input-password',
    onChangeHandler: jest.fn(),
    onBlurValidate: jest.fn(),
  };

  it('should render', () => {
    const wrapper = render(<ControlledInputPassword {...props} />);
    expect(wrapper.baseElement).toMatchSnapshot();
  });

  it('should render value', () => {
    props.value = 'pwd';
    const wrapper = render(<ControlledInputPassword {...props} />);
    expect(wrapper.baseElement).toMatchSnapshot();
  });

  it('should render with errors', () => {
    props.error = 'error';
    const wrapper = render(<ControlledInputPassword {...props} />);
    expect(wrapper.baseElement).toMatchSnapshot();
  });

  it('should toggle visibility on and show password', () => {
    const wrapper = render(<ControlledInputPassword {...props} />);
    const input = wrapper.getByTestId('input-password');
    const visibilityButton = wrapper.getByTestId('toggle-visibility-button');

    act(() => {
      fireEvent.change(input, { target: { value: 'new password' } });
    });

    act(() => {
      fireEvent.click(visibilityButton);
    });

    expect(wrapper.baseElement).toMatchSnapshot();
  });

  it('should toggle visibility on and show password with value', () => {
    props.value = 'pwd';
    const wrapper = render(<ControlledInputPassword {...props} />);
    const input = wrapper.getByTestId('input-password');
    const visibilityButton = wrapper.getByTestId('toggle-visibility-button');

    act(() => {
      fireEvent.change(input, { target: { value: 'new password' } });
    });

    fireEvent.click(visibilityButton);
    expect(wrapper.baseElement).toMatchSnapshot();
  });

  it('should toggle visibility on and show password without value', () => {
    props.value = '';
    const wrapper = render(<ControlledInputPassword {...props} />);
    const visibilityButton = wrapper.getByTestId('toggle-visibility-button');

    fireEvent.click(visibilityButton);
    expect(wrapper.baseElement).toMatchSnapshot();
  });

  it('should toggle visibility off and delete password', () => {
    const wrapper = render(<ControlledInputPassword {...props} />);
    const input = wrapper.getByTestId('input-password');
    const visibilityButton = wrapper.getByTestId('toggle-visibility-button');

    act(() => {
      fireEvent.change(input, { target: { value: '' } });
    });

    fireEvent.click(visibilityButton);
    fireEvent.mouseDown(visibilityButton);
    expect(wrapper.baseElement).toMatchSnapshot();
  });
});
