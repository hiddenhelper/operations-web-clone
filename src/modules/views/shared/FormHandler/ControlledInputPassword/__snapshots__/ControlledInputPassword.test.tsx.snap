// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controlled Input Password should render 1`] = `
<body>
  <div>
    <div
      class="MuiFormControl-root makeStyles-formControl-9 makeStyles-loginLabels-7 "
    >
      <h1
        class="MuiTypography-root MuiTypography-caption MuiTypography-colorSecondary MuiTypography-alignLeft"
      >
        Current Password
         
      </h1>
      <div
        data-testid="confirm-password-wrapper"
      >
        <div
          aria-describedby="confirm-password-password"
          class="MuiFormControl-root MuiTextField-root makeStyles-textField-8 MuiFormControl-fullWidth"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
          >
            <div
              class="MuiInputAdornment-root     makeStyles-loginIcons-6 MuiInputAdornment-positionStart"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                />
              </svg>
            </div>
            <input
              aria-invalid="false"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
              data-testid="input-password"
              name="password"
              placeholder="Current Password"
              type="password"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
            >
              <button
                aria-label="toggle-visibility"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                data-testid="toggle-visibility-button"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    class=" makeStyles-inactiveIcon-3 makeStyles-inactiveIcon-3"
                  >
                    visibility-off.svg
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-11 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-12"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Controlled Input Password should render value 1`] = `
<body>
  <div>
    <div
      class="MuiFormControl-root makeStyles-formControl-23 makeStyles-loginLabels-21 "
    >
      <h1
        class="MuiTypography-root MuiTypography-caption MuiTypography-colorSecondary MuiTypography-alignLeft"
      >
        Current Password
         
      </h1>
      <div
        data-testid="confirm-password-wrapper"
      >
        <div
          aria-describedby="confirm-password-password"
          class="MuiFormControl-root MuiTextField-root makeStyles-textField-22 MuiFormControl-fullWidth"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
          >
            <div
              class="MuiInputAdornment-root makeStyles-filledInput-19    makeStyles-loginIcons-20 MuiInputAdornment-positionStart"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                />
              </svg>
            </div>
            <input
              aria-invalid="false"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
              data-testid="input-password"
              name="password"
              placeholder="Current Password"
              type="password"
              value="pwd"
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
            >
              <button
                aria-label="toggle-visibility"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                data-testid="toggle-visibility-button"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    class=" makeStyles-inactiveIcon-17 makeStyles-filledInput-19"
                  >
                    visibility-off.svg
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-25 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-26"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Controlled Input Password should render with errors 1`] = `
<body>
  <div>
    <div
      class="MuiFormControl-root makeStyles-formControl-37 makeStyles-loginLabels-35 "
    >
      <h1
        class="MuiTypography-root MuiTypography-caption MuiTypography-colorSecondary MuiTypography-alignLeft"
      >
        Current Password
         
      </h1>
      <div
        data-testid="confirm-password-wrapper"
      >
        <div
          aria-describedby="confirm-password-password"
          class="MuiFormControl-root MuiTextField-root makeStyles-textField-36 MuiFormControl-fullWidth"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
          >
            <div
              class="MuiInputAdornment-root   makeStyles-errorIcon-32  makeStyles-loginIcons-34 MuiInputAdornment-positionStart"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                />
              </svg>
            </div>
            <input
              aria-invalid="true"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
              data-testid="input-password"
              name="password"
              placeholder="Current Password"
              type="password"
              value="pwd"
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
            >
              <button
                aria-label="toggle-visibility"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                data-testid="toggle-visibility-button"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    class=" makeStyles-inactiveIcon-31 makeStyles-filledInput-33"
                  >
                    visibility-off.svg
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-39 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-40"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Controlled Input Password should toggle visibility off and delete password 1`] = `
<body>
  <div>
    <div
      class="MuiFormControl-root makeStyles-formControl-93 makeStyles-loginLabels-91 "
    >
      <h1
        class="MuiTypography-root MuiTypography-caption MuiTypography-colorSecondary MuiTypography-alignLeft"
      >
        Current Password
         
      </h1>
      <div
        data-testid="confirm-password-wrapper"
      >
        <div
          aria-describedby="confirm-password-password"
          class="MuiFormControl-root MuiTextField-root makeStyles-textField-92 MuiFormControl-fullWidth"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
          >
            <div
              class="MuiInputAdornment-root   makeStyles-errorIcon-88  makeStyles-loginIcons-90 MuiInputAdornment-positionStart"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                />
              </svg>
            </div>
            <input
              aria-invalid="true"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
              data-testid="input-password"
              name="password"
              placeholder="Current Password"
              type="text"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
            >
              <button
                aria-label="toggle-visibility"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                data-testid="toggle-visibility-button"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    class=" makeStyles-inactiveIcon-87 makeStyles-inactiveIcon-87"
                  >
                    visibility-on.svg
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                >
                  <span
                    class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
                    style="width: 1px; height: 1px; top: -0.5px; left: -0.5px;"
                  >
                    <span
                      class="MuiTouchRipple-child"
                    />
                  </span>
                </span>
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-95 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-96"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Controlled Input Password should toggle visibility on and show password 1`] = `
<body>
  <div>
    <div
      class="MuiFormControl-root makeStyles-formControl-51 makeStyles-loginLabels-49 "
    >
      <h1
        class="MuiTypography-root MuiTypography-caption MuiTypography-colorSecondary MuiTypography-alignLeft"
      >
        Current Password
         
      </h1>
      <div
        data-testid="confirm-password-wrapper"
      >
        <div
          aria-describedby="confirm-password-password"
          class="MuiFormControl-root MuiTextField-root makeStyles-textField-50 MuiFormControl-fullWidth"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
          >
            <div
              class="MuiInputAdornment-root   makeStyles-errorIcon-46  makeStyles-loginIcons-48 MuiInputAdornment-positionStart"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                />
              </svg>
            </div>
            <input
              aria-invalid="true"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
              data-testid="input-password"
              name="password"
              placeholder="Current Password"
              type="text"
              value="pwd"
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
            >
              <button
                aria-label="toggle-visibility"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                data-testid="toggle-visibility-button"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    class=" makeStyles-inactiveIcon-45 makeStyles-filledInput-47"
                  >
                    visibility-on.svg
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-53 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-54"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Controlled Input Password should toggle visibility on and show password with value 1`] = `
<body>
  <div>
    <div
      class="MuiFormControl-root makeStyles-formControl-65 makeStyles-loginLabels-63 "
    >
      <h1
        class="MuiTypography-root MuiTypography-caption MuiTypography-colorSecondary MuiTypography-alignLeft"
      >
        Current Password
         
      </h1>
      <div
        data-testid="confirm-password-wrapper"
      >
        <div
          aria-describedby="confirm-password-password"
          class="MuiFormControl-root MuiTextField-root makeStyles-textField-64 MuiFormControl-fullWidth"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
          >
            <div
              class="MuiInputAdornment-root   makeStyles-errorIcon-60  makeStyles-loginIcons-62 MuiInputAdornment-positionStart"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                />
              </svg>
            </div>
            <input
              aria-invalid="true"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
              data-testid="input-password"
              name="password"
              placeholder="Current Password"
              type="text"
              value="pwd"
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
            >
              <button
                aria-label="toggle-visibility"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                data-testid="toggle-visibility-button"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    class=" makeStyles-inactiveIcon-59 makeStyles-filledInput-61"
                  >
                    visibility-on.svg
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-67 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-68"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Controlled Input Password should toggle visibility on and show password without value 1`] = `
<body>
  <div>
    <div
      class="MuiFormControl-root makeStyles-formControl-79 makeStyles-loginLabels-77 "
    >
      <h1
        class="MuiTypography-root MuiTypography-caption MuiTypography-colorSecondary MuiTypography-alignLeft"
      >
        Current Password
         
      </h1>
      <div
        data-testid="confirm-password-wrapper"
      >
        <div
          aria-describedby="confirm-password-password"
          class="MuiFormControl-root MuiTextField-root makeStyles-textField-78 MuiFormControl-fullWidth"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
          >
            <div
              class="MuiInputAdornment-root   makeStyles-errorIcon-74  makeStyles-loginIcons-76 MuiInputAdornment-positionStart"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                />
              </svg>
            </div>
            <input
              aria-invalid="true"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
              data-testid="input-password"
              name="password"
              placeholder="Current Password"
              type="text"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
            >
              <button
                aria-label="toggle-visibility"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                data-testid="toggle-visibility-button"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    class=" makeStyles-inactiveIcon-73 makeStyles-inactiveIcon-73"
                  >
                    visibility-on.svg
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-81 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-82"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
